storage/mysql/log_storage.go

```diff
@@ -609,11 +609,10 @@ func (t *logTreeTX) GetLeavesByRange(ctx context.Context, start, count int64) ([
 			return nil, fmt.Errorf("got invalid integrate timestamp: %v", err)
 		}
 		ret = append(ret, leaf)
-		wantIndex++
 	}
 
 	if len(ret) == 0 {
-		return nil, status.Errorf(codes.InvalidArgument, "no leaves found in range [%d, %d+%d)", start, start, count)
+		return nil, status.Errorf(codes.InvalidArgument, "no contiguous prefix of leaves found in [%d, %d+%d)", start, start, count)
```

Comment:
I have an impression this shouldn't be an error. For example, if the request is `(start=treeSize, count=100)`, looks reasonable to return an empty slice?

Reply:
Also, this it probably not `InvalidArgument` anymore.